charts/matrix-stack/templates/element-web/deployment.yaml

```diff
@@ -29,9 +29,11 @@ spec:
   template:
     metadata:
       labels:
-        {{- include "element-io.element-web.labels" (dict "root" $ "context" .) | nindent 8 }}
-        k8s.element.io/confighash: "{{ include (print $.Template.BasePath "/element-web/configmap.yaml") $ | sha1sum }}"
-        k8s.element.io/nginxhash: "{{ include (print $.Template.BasePath "/element-web/nginx_configmap.yaml") $ | sha1sum }}"
+        {{- include "element-io.element-web.labels" (dict "root" $ "context" (dict "image" .image "labels" .labels "withChartVersion" false)) | nindent 8 }}
+        k8s.element.io/confighash: "{{( tpl ($.Files.Get "configs/element-web/config.json.tpl") (dict "root" $ "context" .)) | sha1sum }}"
+        k8s.element.io/nginxhash: "{{ (cat ($.Files.Get "configs/element-web/default.conf")
+                                            ($.Files.Get "configs/element-web/http_customisations.conf")
+                                            ($.Files.Get "configs/element-web/security_headers.conf"))  | sha1sum }}"
```

Comment:
I think we should have a named template (e.g. `element-io.synapse.secret-data` exists) that does all these `$.Files.Get`. That gives us:
* Ensures if we add a new file to the `Secret`/`ConfigMap` these hash labels will get it without us having to remember
* We rename the file in the `Secret`/`ConfigMap` that also changes the hash label